src/Lib.hs

```diff
@@ -95,27 +95,37 @@ fetchAgents = do
 
 processTicket :: TicketId -> App (Maybe ZendeskResponse)
 processTicket tId = do
+    printText <- asksIOLayer iolPrintText
+    -- Printing id before inspecting the ticket so that when the process stops by the
+    -- corrupted log file, we know which id to blacklist.
+    printText $ "Analyzing ticket id: " <> show (getTicketId tId)
 
     -- We first fetch the function from the configuration
     getTicketInfo       <- asksZendeskLayer zlGetTicketInfo
-    printText           <- asksIOLayer iolPrintText
-
-    printText $ "Processing a ticket: " <> show (getTicketId tId)
-
     mTicketInfo         <- getTicketInfo tId
+    
     getTicketComments   <- asksZendeskLayer zlGetTicketComments
-    comments      <- getTicketComments tId
+    comments            <- getTicketComments tId
+    
     let attachments = getAttachmentsFromComment comments
     let ticketInfo  = fromMaybe (error "No ticket info") mTicketInfo
     zendeskResponse     <- getZendeskResponses comments attachments ticketInfo
-    postTicketComment   <- asksZendeskLayer zlPostTicketComment
-    whenJust zendeskResponse postTicketComment
 
-    printText "Process finished, please see the following url"
-    printText $ "https://iohk.zendesk.com/agent/tickets/" <> show (getTicketId tId)
+    postTicketComment   <- asksZendeskLayer zlPostTicketComment
+    whenJust zendeskResponse $ \response -> do
+        printZendeskResponse response
+        postTicketComment response
 
     pure zendeskResponse
 
+printZendeskResponse :: ZendeskResponse -> App ()
+printZendeskResponse zendeskResponse = do
+    let ticketTags = zrTags zendeskResponse
+    printText <- asksIOLayer iolPrintText
```

Comment:
Move this out of this function to keep it pure. Call it `printZendeskResponse` and fetch `iolPrintText` and print it in the function above.

Reply:
Nice point!